Server/Services/RevolutionaryDiscordBotService.cs

```diff
@@ -269,13 +285,16 @@ private async Task ClientReady()
                 await guild.CreateApplicationCommandAsync(BuildLanguageCommand().Build());
                 await guild.CreateApplicationCommandAsync(BuildWikiCommand().Build());
                 await guild.CreateApplicationCommandAsync(BuildReleasesCommand().Build());
+                await guild.CreateApplicationCommandAsync(BuildDsucCommand().Build());
             }
             catch (HttpException e)
             {
                 logger.LogError(e, "Failed to register one or more guild commands");
             }
         }
 
+        await databaseReadWriteLock.WaitAsync();
+
         // If commands are changed the version numbers here *must* be updated
         bool changes = await RegisterGlobalCommandIfRequired(BuildProgressCommand(), 2);
```

Comment:
This whole section must be in a `try ... finally` block to be safe regarding this lock.